lib/bn.js

```diff
@@ -450,6 +450,11 @@ BN.prototype.toString = function toString(base, padding) {
   }
 };
 
+BN.prototype.toNumber = function toNumber() {
+  assert(this.bitLength() <= 53, 'Number can only safely store up to 53 bits');
+  return parseInt(this.toString());
```

Comment:
That will allow bitLength()>53 numbers to be wrongly transformed into number. JS won't complain, see http://stackoverflow.com/questions/307179/what-is-javascripts-highest-integer-value-that-a-number-can-go-to-without-losin